src/js/actions/history.js

```diff
@@ -279,6 +283,7 @@ define(function (require, exports) {
     };
     onReset.reads = [];
     onReset.writes = [];
+    onReset.modal = true;
```

Comment:
It seems like part of the onReset routine should be to end any modal states. Do you know if that is currently true? Was there a particular motivation for marking this action as modal safe?